packages/hoppscotch-app/helpers/functional/debug.ts

```diff
@@ -7,3 +10,20 @@ export const trace = <T>(x: T) => {
   console.log(x)
   return x
 }
+
+/**
+ * Logs the annotated current value and returns the same value
+ * @param name The name of the log
+ * @curried_param `x` The value to log
+ * @returns The parameter `x` passed to this
+ */
+export const namedTrace =
+  (name: string) =>
+  <T>(x: T) => {
+    const printFn = (val?: T | string) => console.log(`${name}: ${val ?? x}`)
+    pipe(
+      O.tryCatch(() => JSON.stringify(x)),
+      O.match(printFn, printFn)
+    )
```

Comment:
Sometimes it just printed `[object Object]`, so I did that.

It also had a filter to check if it is `not an array` and of type `object`, but tryCatch handled it regardless and there was no performance requirement since we were debugging, so I removed that O.filter.

Reply:
@AndrewBastin

Reply:
> Sometimes it just printed [object Object], so I did that.

My suggestion won't. Try it 😉 
NOTE that it is ```console.log(`${name}: `, x)``` not ```console.log(`${name}: ${x}`)```
(x is a second parameter)